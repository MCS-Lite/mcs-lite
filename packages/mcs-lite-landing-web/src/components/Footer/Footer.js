import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import P from 'mcs-lite-ui/lib/P';

const Wrapper = styled.div`
  background-color: ${props => props.theme.color.black};
  text-align: center;
  padding: 6px 0;
`;

const Footer = ({ getMessages: t }) =>
  <Wrapper>
    <P color="white">Â© 2017 {t('desc')}</P>
  </Wrapper>;

Footer.displayName = 'Footer';
Footer.propTypes = {
  // React-intl I18n
  getMessages: PropTypes.func.isRequired,
};

export default Footer;
