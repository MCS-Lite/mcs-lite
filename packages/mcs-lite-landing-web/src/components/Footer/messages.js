import { defineMessages } from 'react-intl';

const messages = defineMessages({
  'Footer.desc': {
    id: 'Footer.desc',
    defaultMessage: 'MediaTek Inc. All Rights Reserved.',
  },
});

export default messages;
