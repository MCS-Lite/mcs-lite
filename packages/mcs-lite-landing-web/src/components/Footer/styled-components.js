import styled from 'styled-components';
import { Column, withBreakpoints } from 'hedron';

export const Container = styled.footer`
  background-color: ${props => props.theme.color.black};
`;

export const StyledColumn = styled(Column)`
  padding-top: 10px;
  padding-bottom: 10px;
`;

export const IconWrapper = styled.div`
  display: flex;

  a {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  a + a {
    margin-left: 20px;
  }

  img {
    height: 32px;
    width: 32px;
    color: ${props => props.theme.color.white};
  }
`;

export const FakeIcon = styled.div`
  height: 32px;
  width: 32px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.color.white};
  border-radius: 4px;
  color: ${props => props.theme.color.black};
  text-decoration: none;
`;

export const RWDWrapper = withBreakpoints(styled.div`
  display: flex;
  align-items: center;
  text-align: center;
  flex-direction: column;

  ${IconWrapper} {
    margin-bottom: 5px;
  }

  @media (min-width: ${props => props.breakpoints.sm}px) {
    justify-content: space-between;
    flex-direction: row-reverse;

    ${IconWrapper} {
      margin-bottom: 0;
    }
  }
`);
