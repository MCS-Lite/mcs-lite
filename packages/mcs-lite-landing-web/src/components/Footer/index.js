import { withGetMessages } from 'react-intl-inject-hoc';
import { compose, pure } from 'recompose';
import Footer from './Footer';
import messages from './messages';

export default compose(pure, withGetMessages(messages, 'Footer'))(Footer);
