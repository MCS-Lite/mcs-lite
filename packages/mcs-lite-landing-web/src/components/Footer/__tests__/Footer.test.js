import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from '../Footer';

it('should renders <Footer> correctly', () => {
  const wrapper = shallow(<Footer getMessages={t => t} />);

  const tree = toJson(wrapper);

  expect(tree).toMatchStyledComponentsSnapshot();
});
