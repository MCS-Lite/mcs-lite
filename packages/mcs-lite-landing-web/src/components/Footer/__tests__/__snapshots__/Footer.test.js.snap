// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <Footer> correctly 1`] = `
<styled.footer>
  <Styled(PageContainer)
    width="960px"
  >
    <Styled(RowContainer)>
      <Styled(Breakpoints)
        xs={12}
      >
        <Breakpoints>
          <styled.div>
            <A
              color="primary"
              href="https://www.facebook.com/groups/1651382875112603"
              rel="noreferrer noopener"
              target="_blank"
            >
              <LazyloadOnce
                component="div"
                height={32}
                waypointConfig={
                  Object {
                    "bottomOffset": -500,
                    "fireOnRapidScroll": true,
                    "topOffset": -500,
                  }
                }
              >
                <LoadableComponent />
              </LazyloadOnce>
            </A>
            <A
              color="primary"
              href="https://github.com/MCS-Lite"
              rel="noreferrer noopener"
              target="_blank"
            >
              <LazyloadOnce
                component="div"
                height={32}
                waypointConfig={
                  Object {
                    "bottomOffset": -500,
                    "fireOnRapidScroll": true,
                    "topOffset": -500,
                  }
                }
              >
                <LoadableComponent />
              </LazyloadOnce>
            </A>
          </styled.div>
          <P
            color="white"
          >
            Â© 
            2017
             
            desc
          </P>
        </Breakpoints>
      </Styled(Breakpoints)>
    </Styled(RowContainer)>
  </Styled(PageContainer)>
</styled.footer>
`;
